src/devices/sound/pokey.cpp

```diff
@@ -1124,36 +1179,32 @@ void pokey_device::vol_init()
 
 }
 
-void pokey_device::poly_init_4_5(uint32_t *poly, int size, int xorbit, int invert)
+void pokey_device::poly_init_4_5(uint32_t *poly, int size)
 {
+	LOG_POLY(("poly %d\n", size));
+
 	int mask = (1 << size) - 1;
-	int i;
 	uint32_t lfsr = 0;
 
-	LOG_POLY(("poly %d\n", size));
-	for( i = 0; i < mask; i++ )
+	int const xorbit = size - 1;
+	for (int i = 0; i < mask; i++)
 	{
-		/* calculate next bit */
-		int in = !((lfsr >> 0) & 1) ^ ((lfsr >> xorbit) & 1);
-		lfsr = lfsr >> 1;
-		lfsr = (in << (size-1)) | lfsr;
-		*poly = lfsr ^ invert;
-		LOG_POLY(("%05x: %02x\n", i, *poly));
+		lfsr = (lfsr << 1) | (~((lfsr >> 2) ^ (lfsr >> xorbit)) & 1);
+		*poly = lfsr & mask;
 		poly++;
 	}
 }
 
 void pokey_device::poly_init_9_17(uint32_t *poly, int size)
 {
-	int mask = (1 << size) - 1;
-	int i;
-	uint32_t lfsr =mask;
-
 	LOG_RAND(("rand %d\n", size));
 
+	int mask = (1 << size) - 1;
+	uint32_t lfsr = mask;
```

Comment:
You can use `util::make_bitmask` to make a mask with a specified number of bits set.